File: solutions/1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.go
func kLengthApart(nums []int, k int) bool {
	start := false
	gap := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] == 1 {
			if start && gap < k {
				return false
			}
			start = true
			gap = 0
		} else if start && nums[i] == 0 {
			gap++
		}
	}
	return true
}

File: solutions/1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.rs
impl Solution {
    pub fn k_length_apart(nums: Vec<i32>, k: i32) -> bool {
        nums.iter()
            .enumerate()
            .fold((-0xFFFFFFF, true), |(mut l, mut r), (i, x)| {
                if x.eq(&1) {
                    if !l.eq(&-0xFFFFFFF) {
                        if i as i32 - l - 1 < k { r = false }                   
                    } l = i as i32;
                } (l, r)
            }).1
    }
}